### PR TITLE
build: split material ng-update schematics to a separate bundle

### PR DESCRIPTION
**Pull request by wagnermaciel**
* split material ng-update schematics to a separate bundle
* delete `resolveBazelPath` and uses of `__dirname` and replace w/ `runfiles.resolve...`
* export gesture config template as a variable instead of having to use fs readFile

### CODE REVIEW

File: src/material/schematics/ng-update/test-cases/misc/class-inheritance.spec.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
-import {createTestCaseSetup, resolveBazelPath} from '@angular/cdk/schematics/testing';
+import {runfiles} from '@bazel/runfiles';
+import {createTestCaseSetup} from '@angular/cdk/schematics/testing';
 import {MIGRATION_PATH} from '../../../paths';
 
 describe('class inheritance misc checks', () => {
   describe('v6 class which extends MatFormFieldControl', () => {
     it('should report if class does not declare "shouldLabelFloat"', async () => {
+      console.log('MIGRATION PATH: ---> ', MIGRATION_PATH);
```
Comment (devversion) [inline src/material/schematics/ng-update/test-cases/misc/class-inheritance.spec.ts:8]: Leftover from debugging

File: src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export function isHammerJsUsedInTemplate(html: string): {
       }
     });
   };
-  visitNodes(document.childNodes);
+  visitNodes((document as any).childNodes as parse5.ChildNode[]);
```
Comment (devversion) [inline src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts:90]: Is there a reason the type is unspecified?

File: src/material/schematics/BUILD.bazel [hunk 3]
```diff
@@ -41,6 +53,23 @@ ts_library(
...
 
+ts_library(
+    name = "paths",
+    testonly = True,
+    srcs = ["paths.ts"],
+    data = [
+        ":collection_assets",
+    ],
+    # Schematics do not need to run in browsers and can use `commonjs`
+    # as format instead the default `umd` format.
```
Comment (devversion) [inline src/material/schematics/BUILD.bazel:64]: Should be I think:

```
    # Schematics can not yet run in ESM module. For now we continue to use CommonJS.
    # TODO(ESM): remove this once the Angular CLI supports ESM schematics.
```

**Review by devversion — APPROVED** [review]

File: src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts (hunk 2 above)
Comment (wagnermaciel) [reply src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts:90]: I swapped to the alternative: `visitNodes((document as unknown as parse5.Document).childNodes);`

Comment (wagnermaciel) [reply src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts]: Oh, I was getting typescript errors about it complaining that `document` was of type `unknown`

Comment (wagnermaciel) [reply src/material/schematics/ng-update/migrations/hammer-gestures-v9/hammer-template-check.ts]: I think the exact error was:
`ts2571: object is of type 'unknown'`

**Pull request merged**